commits.go

```diff
@@ -35,23 +35,24 @@ type CommitsService struct {
 //
 // GitLab API docs: https://docs.gitlab.com/ce/api/commits.html
 type Commit struct {
-	ID             string           `json:"id"`
-	ShortID        string           `json:"short_id"`
-	Title          string           `json:"title"`
-	AuthorName     string           `json:"author_name"`
-	AuthorEmail    string           `json:"author_email"`
-	AuthoredDate   *time.Time       `json:"authored_date"`
-	CommitterName  string           `json:"committer_name"`
-	CommitterEmail string           `json:"committer_email"`
-	CommittedDate  *time.Time       `json:"committed_date"`
-	CreatedAt      *time.Time       `json:"created_at"`
-	Message        string           `json:"message"`
-	ParentIDs      []string         `json:"parent_ids"`
-	Stats          *CommitStats     `json:"stats"`
-	Status         *BuildStateValue `json:"status"`
-	LastPipeline   *PipelineInfo    `json:"last_pipeline"`
-	ProjectID      int              `json:"project_id"`
-	WebURL         string           `json:"web_url"`
+	ID             string                  `json:"id"`
+	ShortID        string                  `json:"short_id"`
+	Title          string                  `json:"title"`
+	AuthorName     string                  `json:"author_name"`
+	AuthorEmail    string                  `json:"author_email"`
+	AuthoredDate   *time.Time              `json:"authored_date"`
+	CommitterName  string                  `json:"committer_name"`
+	CommitterEmail string                  `json:"committer_email"`
+	CommittedDate  *time.Time              `json:"committed_date"`
+	CreatedAt      *time.Time              `json:"created_at"`
+	Message        string                  `json:"message"`
+	ParentIDs      []string                `json:"parent_ids"`
+	Stats          *gitlab.CommitStats     `json:"stats"`
+	Status         *gitlab.BuildStateValue `json:"status"`
+	LastPipeline   *gitlab.PipelineInfo    `json:"last_pipeline"`
+	ProjectID      int                     `json:"project_id"`
+	WebURL         string                  `json:"web_url"`
+	Trailers       map[string]string       `json:"trailers"`
```

Comment:
Where can I see that data is returned for the trailers field? Cannot find it in the docs...

Reply:
<details>
  <summary markdown="span">Example `GET $CI_API_V4_URL/projects/:id/repository/commits?trailers=true`</summary>
```json
[
    {
        "id": "a90105c61a1a646127be97eac670379b74c721ec",
        "short_id": "a90105c6",
        "created_at": "2022-04-17T22:03:27.000+00:00",
        "parent_ids": [
            "40083eae4542e45af406f96a9733375d50f2cfe4",
            "c10ff20f2794270a4f6f8c18129a3e3cd2f2c5d8"
        ],
        "title": "Merge branch 'Nickolena.Fisher-main-patch-69867' into 'main'",
        "message": "Merge branch 'Nickolena.Fisher-main-patch-69867' into 'main'\n\nAdd test file 3\n\nSee merge request Nickolena.Fisher/release-tools-test!3",
        "author_name": "Nickolena.Fisher",
        "author_email": "nickolena.fisher@envestnet.com",
        "authored_date": "2022-04-17T22:03:27.000+00:00",
        "committer_name": "Nickolena.Fisher",
        "committer_email": "nickolena.fisher@envestnet.com",
        "committed_date": "2022-04-17T22:03:27.000+00:00",
        "trailers": {},
        "web_url": "/-/commit/a90105c61a1a646127be97eac670379b74c721ec"
    },
    {
        "id": "c10ff20f2794270a4f6f8c18129a3e3cd2f2c5d8",
        "short_id": "c10ff20f",
        "created_at": "2022-04-17T22:03:13.000+00:00",
        "parent_ids": [
            "40083eae4542e45af406f96a9733375d50f2cfe4"
        ],
        "title": "Add test file 3",
        "message": "Add test file 3\n\nChangelog: added",
        "author_name": "Nickolena.Fisher",
        "author_email": "nickolena.fisher@envestnet.com",
        "authored_date": "2022-04-17T22:03:13.000+00:00",
        "committer_name": "Nickolena.Fisher",
        "committer_email": "nickolena.fisher@envestnet.com",
        "committed_date": "2022-04-17T22:03:13.000+00:00",
        "trailers": {
            "Changelog": "added"
        },
        "web_url": "/-/commit/c10ff20f2794270a4f6f8c18129a3e3cd2f2c5d8"
    },
    {
        "id": "40083eae4542e45af406f96a9733375d50f2cfe4",
        "short_id": "40083eae",
        "created_at": "2022-04-17T22:00:49.000+00:00",
        "parent_ids": [
            "ca00ca94d0e2574b355968e886ce72394a7b5c6e",
            "d91d61cc58337fa8e37a1f6dd7f5ec939c5d5ef5"
        ],
        "title": "Merge branch 'Nickolena.Fisher-main-patch-18097' into 'main'",
        "message": "Merge branch 'Nickolena.Fisher-main-patch-18097' into 'main'\n\nFix test file\n\nSee merge request Nickolena.Fisher/release-tools-test!2",
        "author_name": "Nickolena.Fisher",
        "author_email": "nickolena.fisher@envestnet.com",
        "authored_date": "2022-04-17T22:00:49.000+00:00",
        "committer_name": "Nickolena.Fisher",
        "committer_email": "nickolena.fisher@envestnet.com",
        "committed_date": "2022-04-17T22:00:49.000+00:00",
        "trailers": {},
        "web_url": "https://gitlab.env.io/Nickolena.Fisher/release-tools-test/-/commit/40083eae4542e45af406f96a9733375d50f2cfe4"
    },
    {
        "id": "d91d61cc58337fa8e37a1f6dd7f5ec939c5d5ef5",
        "short_id": "d91d61cc",
        "created_at": "2022-04-17T22:00:37.000+00:00",
        "parent_ids": [
            "ca00ca94d0e2574b355968e886ce72394a7b5c6e"
        ],
        "title": "Fix test file",
        "message": "Fix test file\n\nChangelog: fixed",
        "author_name": "Nickolena.Fisher",
        "author_email": "nickolena.fisher@envestnet.com",
        "authored_date": "2022-04-17T22:00:37.000+00:00",
        "committer_name": "Nickolena.Fisher",
        "committer_email": "nickolena.fisher@envestnet.com",
        "committed_date": "2022-04-17T22:00:37.000+00:00",
        "trailers": {
            "Changelog": "fixed"
        },
        "web_url": "/-/commit/d91d61cc58337fa8e37a1f6dd7f5ec939c5d5ef5"
    }
]
```
</details>